src/include/ipc4/notification.h

```diff
@@ -168,3 +229,26 @@ static inline void ipc4_notification_watchdog_init(struct ipc4_watchdog_timeout_
 	notif->primary.r.rsp = SOF_IPC4_MESSAGE_DIR_MSG_REQUEST;
 	notif->primary.r.msg_tgt = SOF_IPC4_MESSAGE_TARGET_FW_GEN_MSG;
 }
+
+/**
+ * \brief Input data payload is reserved field in parent technical spec which can be easily
+ * extendable if needed by specific resource event types in the future. For backward compatibility
+ * the size of this stucture is 0.
+ */
+union ipc4_resource_event_data {
+	/* Raw data */
+	uint32_t dws[0];
+};
+
+struct ipc4_resource_event_data_notification {
+	/* Type of originator (see sof_ipc4_resource_type) */
+	uint32_t resource_type;
+	/* ID of resource firing event */
+	uint32_t resource_id;
+	/* Type of fired event (see sof_ipc4_resource_event_type enum) */
+	uint32_t event_type;
+	/* Explicit alignment as ipc4_resource_event_data contains 64bit fields and needs alignment */
+	uint32_t reserved0;
```

Comment:
dont we need to use ```reserved0``` for size ? (or even part of ```reserved0``` like 16bits)

Reply:
I think we will be able to deduce all the sizes by the event_type and appropriate size of future structures for each one. I think reserved0 as a size is not necessary.

Reply:
using types implies that all ingredients in FW/SW and tooling need to know about the type and map it to size (even just to pass that data through to the next layer in the stack - e.g. FW passing event to userspace means kernel needs to map the type to size). 
Using all or some of reserved0 allows SW not to care about type (when there is no direct data usage) and just memcpy() the data to the next layer.

Reply:
@makarukp thanks for explaining this only used for resource info and not for general purpose event messaging to host. All good to go.